src/lib/netlist/macro/nlm_ttl74xx_lib.cpp

```diff
@@ -3159,6 +3320,61 @@ static TRUTH_TABLE(TTL_74139_GATE, 3, 4, "")
 	TT_FAMILY("74XX")
 }
 
+static TRUTH_TABLE(TTL_74147_GATE, 9, 4, "")
+{
+	TT_HEAD("1,2,3,4,5,6,7,8,9|D,C,B,A")
+	TT_LINE("1,1,1,1,1,1,1,1,1|1,1,1,1|10")
+	TT_LINE("X,X,X,X,X,X,X,X,0|0,1,1,0|10")
+	TT_LINE("X,X,X,X,X,X,X,0,1|0,1,1,1|10")
+	TT_LINE("X,X,X,X,X,X,0,1,1|1,0,0,0|10")
+	TT_LINE("X,X,X,X,X,0,1,1,1|1,0,0,1|10")
+	TT_LINE("X,X,X,X,0,1,1,1,1|1,0,1,0|10")
+	TT_LINE("X,X,X,0,1,1,1,1,1|1,0,1,1|10")
+	TT_LINE("X,X,0,1,1,1,1,1,1|1,1,0,0|10")
+	TT_LINE("X,0,1,1,1,1,1,1,1|1,1,0,1|10")
+	TT_LINE("0,1,1,1,1,1,1,1,1|1,1,1,0|10")
+	TT_FAMILY("74XX")
+}
+
+static TRUTH_TABLE(TTL_74148_GATE, 9, 5, "")
+{
+	TT_HEAD("EI,0,1,2,3,4,5,6,7|A2,A1,A0,GS,EO")
+	TT_LINE( "1,X,X,X,X,X,X,X,X|1,1,1,1,1|10")
+	TT_LINE( "0,1,1,1,1,1,1,1,1|1,1,1,1,0|10")
+	TT_LINE( "0,X,X,X,X,X,X,X,0|0,0,0,0,1|10")
+	TT_LINE( "0,X,X,X,X,X,X,0,1|0,0,1,0,1|10")
+	TT_LINE( "0,X,X,X,X,X,0,1,1|0,1,0,0,1|10")
+	TT_LINE( "0,X,X,X,X,0,1,1,1|0,1,1,0,1|10")
+	TT_LINE( "0,X,X,X,0,1,1,1,1|1,0,0,0,1|10")
+	TT_LINE( "0,X,X,0,1,1,1,1,1|1,0,1,0,1|10")
+	TT_LINE( "0,X,0,1,1,1,1,1,1|1,1,0,0,1|10")
+	TT_LINE( "0,0,1,1,1,1,1,1,1|1,1,1,0,1|10")
+	TT_FAMILY("74XX")
+}
+
+static TRUTH_TABLE(TTL_74151_GATE, 12, 2, "")
+{
+	TT_HEAD("C,B,A,G,D0,D1,D2,D3,D4,D5,D6,D7|Y,W")
+	TT_LINE( "X,X,X,1,X,X,X,X,X,X,X,X|0,1|38,30")
+	TT_LINE( "0,0,0,0,1,X,X,X,X,X,X,X|1,0|38,30")
+	TT_LINE( "0,0,0,0,0,X,X,X,X,X,X,X|0,1|38,30")
+	TT_LINE( "0,0,1,0,X,1,X,X,X,X,X,X|1,0|38,30")
+	TT_LINE( "0,0,1,0,X,0,X,X,X,X,X,X|0,1|38,30")
+	TT_LINE( "0,1,0,0,X,X,1,X,X,X,X,X|1,0|38,30")
+	TT_LINE( "0,1,0,0,X,X,0,X,X,X,X,X|0,1|38,30")
+	TT_LINE( "0,1,1,0,X,X,X,1,X,X,X,X|1,0|38,30")
+	TT_LINE( "0,1,1,0,X,X,X,0,X,X,X,X|0,1|38,30")
+	TT_LINE( "1,0,0,0,X,X,X,X,1,X,X,X|1,0|38,30")
+	TT_LINE( "1,0,0,0,X,X,X,X,0,X,X,X|0,1|38,30")
+	TT_LINE( "1,0,1,0,X,X,X,X,X,1,X,X|1,0|38,30")
+	TT_LINE( "1,0,1,0,X,X,X,X,X,0,X,X|0,1|38,30")
+	TT_LINE( "1,1,0,0,X,X,X,X,X,X,1,X|1,0|38,30")
+	TT_LINE( "1,1,0,0,X,X,X,X,X,X,0,X|0,1|38,30")
+	TT_LINE( "1,1,1,0,X,X,X,X,X,X,X,1|1,0|38,30")
+	TT_LINE( "1,1,1,0,X,X,X,X,X,X,X,0|0,1|38,30")
+	TT_FAMILY("74XX")
```

Comment:
Why are you using worst-case propagation delays for the 74LS version on this chip?